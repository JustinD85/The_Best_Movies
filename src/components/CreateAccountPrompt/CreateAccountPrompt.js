import React from 'react'
import { Link } from 'react-router-dom'
import './CreateAccountPrompt.scss'

const CreateAccountPrompt = (props) => {
  const { setShouldPromptLogin, setPopup } = props;
  setPopup(true)

  return <div className='CreateAccountPrompt'>
    <p className='create-account-prompt'>In order to <span role='img' aria-label='star emoji'>‚≠ê</span> your favorite movies, you need an account. Would you like to create one?</p>
    <hr/>
    <Link to='/create-user' onClick={() => setShouldPromptLogin(false)}>YES</Link>
    <button className='no-btn' onClick={() => { setShouldPromptLogin(false); setPopup(false) }}>NO</button>
    <hr/>    
    <p className='account-notice'>(If you already have an account, you can login from the home page!)</p>
  </div >
}

export default CreateAccountPrompt