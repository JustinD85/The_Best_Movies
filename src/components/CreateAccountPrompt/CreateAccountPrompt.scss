.CreateAccountPrompt {
  height: 42%;
  width: 50%;
  z-index: 2;
  font-size: 30px;
  position: absolute;
  color: #ECEBEE;
  padding: 7%;
  border-radius: 25px;
  background: rgba(0, 0, 0, 0.87);
  left: 23%;
  top: 10%;

  .no-btn {
    background: transparent;
    color: #ECEBEE;
    border: none;
    border-bottom: 3px solid transparent;
    margin: 11% 0% 0% 26%;

    &:hover {
      cursor: pointer;
      border-bottom: 3px solid #55E4F5;
    }
  }

  a {
    text-decoration: none;
    color: #ECEBEE;
    border-bottom: 3px solid transparent;

    &:hover {
      cursor: pointer;
      border-bottom: 3px solid #55E4F5;
    }
  }
}