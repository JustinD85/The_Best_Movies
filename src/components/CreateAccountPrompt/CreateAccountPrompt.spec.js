import React from 'react'
import CreateAccountPrompt from './CreateAccountPrompt'
import { shallow } from 'enzyme'

describe('CreateAccountPrompt', () => {
  let setLoginPromptMock
  let setPopupMock
  let wrapper

  beforeEach(() => {
    window.fetch = jest.fn()
    setLoginPromptMock = jest.fn()
    setPopupMock = jest.fn()
    wrapper = shallow(<CreateAccountPrompt
      setLoginPrompt={setLoginPromptMock}
      setPopup={setPopupMock}
    />)
  })

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('should call setLoginPrompt when component mounts', () => {
    expect(setPopupMock).toHaveBeenCalledTimes(1)
  })

  it('should call setPopup and setLoginPrompt when clicked', () => {
    wrapper.find('button').simulate('click')
    expect(setPopupMock).toHaveBeenCalledTimes(2)
    expect(setLoginPromptMock).toHaveBeenCalledTimes(1)
  })

  it('should call setPopup and setLoginPrompt when clicked', () => {
    wrapper.find('Link').simulate('click')
    expect(setPopupMock).toHaveBeenCalledTimes(1)
  })
})