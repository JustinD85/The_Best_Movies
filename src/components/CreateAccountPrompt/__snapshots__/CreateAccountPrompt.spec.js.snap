// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateAccountPrompt should match snapshot 1`] = `
<div
  className="CreateAccountPrompt"
>
  <p>
    Would you like to create an account?
  </p>
  <Link
    onClick={[Function]}
    replace={false}
    to="/create-user"
  >
    YES
  </Link>
  <button
    className="no-btn"
    onClick={[Function]}
  >
    NO
  </button>
</div>
`;
